WooCommerce/Classes/ViewRelated/Dashboard/Settings/SettingsViewController.swift

```diff
@@ -112,37 +131,96 @@ private extension SettingsViewController {
         tableView.delegate = self
     }
 
-    func refreshResultsController() {
-        try? resultsController.performFetch()
-        sites = resultsController.fetchedObjects
-    }
+    /// Set up observation of the results controllers, so that when new data arrives
+    /// the view can be refreshed, and then perform the initial fetch from storage.
+    ///
+    private func configureResultsControllers(onReload: @escaping () -> Void) {
+        sitesResultsController.onDidChangeContent = {
+            onReload()
+        }
 
-    func checkAvailabilityForPayments(onCompletion: @escaping () -> Void) {
-        guard let siteID = self.siteID else {
-            canCollectPayments = false
-            onCompletion()
+        sitesResultsController.onDidResetContent = { [weak self] in
+            guard let self = self else {
+                return
+            }
+
+            self.refetchAllResultsControllers()
+            onReload()
+        }
+
+        try? sitesResultsController.performFetch()
+
+        guard paymentGatewayAccountsResultsController != nil else {
             return
         }
 
-        let action = WCPayAction.loadAccount(siteID: siteID) { [weak self] result in
+        paymentGatewayAccountsResultsController?.onDidChangeContent = {
+            onReload()
+        }
+
+        paymentGatewayAccountsResultsController?.onDidResetContent = { [weak self] in
             guard let self = self else {
                 return
             }
+            self.refetchAllResultsControllers()
+            onReload()
+        }
 
-            switch result {
-            case .failure:
-                self.canCollectPayments = false
-            case .success(let account):
-                self.canCollectPayments = ServiceLocator.featureFlagService.isFeatureFlagEnabled(.cardPresentPayments) &&
-                    account.isCardPresentEligible
-            }
+        try? paymentGatewayAccountsResultsController?.performFetch()
+    }
 
-            onCompletion()
+    /// Refetching all the results controllers is necessary after a storage reset in `onDidResetContent` callback and before reloading UI that
+    /// involves more than one results controller.
+    ///
+    private func refetchAllResultsControllers() {
+        try? sitesResultsController.performFetch()
+        guard paymentGatewayAccountsResultsController != nil else {
+            return
+        }
+        try? paymentGatewayAccountsResultsController?.performFetch()
+    }
+
+    /// Ask the PaymentGatewayAccountStore to loadAccounts from the network and update storage
+    ///
+    private func loadPaymentGatewayAccounts() {
+        guard let siteID = self.siteID else {
+            return
         }
 
+        /// No need for a completion here. We will be notified of storage changes in `onDidChangeContent`
+        ///
+        let action = PaymentGatewayAccountAction.loadAccounts(siteID: siteID) {_ in
```

Comment:
Maybe we don't need a completion block for the loadAccounts method at all? After all we are not going to act on potential errors, and, as mentioned in the comment, we'd get notified in onDidChangeContent